pkgs/test/substitute/default.nix

```diff
@@ -0,0 +1,97 @@
+{ substitute, testers, runCommand }:
+let
+  # Ofborg doesn't allow any traces on stderr,
+  # so mock `lib` to not trace warnings,
+  # because substitute gives a deprecation warning
+  substituteSilent = substitute.override (prevArgs: {
+    lib = prevArgs.lib.extend (finalLib: prevLib: {
+      trivial = prevLib.trivial // {
+        warn = msg: value: value;
+      };
+    });
+  });
+in {
+
+  substitutions = testers.testEqualContents {
+    assertion = "substitutions-spaces";
+    actual = substitute {
+      src = builtins.toFile "source" ''
+        Hello world!
+      '';
+      substitutions = [
+        "--replace-fail"
```

Comment:
Unrelated to this patch: I think `--replace-fail` is what most of the other uses of `--replace` wanted to use. Asking to replace something that doesn't exist is the sign of a rotted replacement!

Reply:
Agreed! I'd rather not bother trying to verify those for now.